pkg/storage/read_as_of_iterator.go

```diff
@@ -94,40 +103,127 @@ func (f *ReadAsOfIterator) UnsafeValue() []byte {
 
 // HasPointAndRange implements SimpleMVCCIterator.
 func (f *ReadAsOfIterator) HasPointAndRange() (bool, bool) {
-	panic("not implemented")
+	// the ReadAsOfIterator only surfaces point keys; therefore hasPoint is always
+	// true, unless the iterator is invalid.
+	return f.valid, f.hasRange && f.valid
```

Comment:
Why do we ever return `true` for range keys here? Shouldn't this iterator only surface live point keys? The same goes for the other range key methods. I suspect this will simplify the iterator significantly.

Reply:
heh, good point. Updated everything.